### ISSUE TITLE
Output screengrab video in headless mode

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#19553]
It might be useful to capture video in headless mode, in the same way as we are currently capturing screenshots. This came up in conversation with @cvan about automating builds of embedded servo apps, since this would allow automated build of demo videos.

cc @glennw

**Comment by frol** [comment]
Wow, perfect timing for me! I was literally searching on the internet for efficient solutions to this problem (I tried capturing real browser tabs with screen/window recorders, e.g. ffmpeg, OBS but that puts a ton of load on my CPU, so when I tried to record 4 videos at a time on my Core i7, it just failed to record at all). And then I remembered about Servo and here I am opening the issues tab and the latest issue is exactly what I am looking for... It is unfortunate that it is only a proposal at this point in time, but I will definitely follow this issue and hope it will be implemented one day!

P.S. I am sorry for the offtopic.

**Comment by cvan** [comment]
@asajeffrey thanks for stopping by to introduce yourself and chat about embeddability. I learned a lot, and I'm glad we have folks such as y'all working on these hard, but important (IMO), problems. :+1:

What's the support in Servo for the [`MediaRecorder`](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) and/or [`OffscreenCanvas`](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) APIs? (I've used `MediaRecorder` with great success in [previous projects](https://github.com/cvan/cancorder/blob/a1b34fc/content_script.js#L181-L197).)

If those APIs are available, I can help with code snippets, tooling, CI. Let me know.

**Comment by asajeffrey** [comment]
@cvan I don't think we have support for MediaRecorder. @nox was most recently looking at the media stack, and might know if we have plans to implement it.

**Comment by jdm** [comment]
@cvan There is an unfinished implementation of part of OffscreenCanvas in an open PR.

**Comment by kimown** [comment]
I want to record CSS3 animation in headless, it's also useful for testing purpose, does this feature possible in servo?

**Comment by jdm** [comment]
Our headless implementation can only take individual screenshots, not video.

**Comment by kimown** [comment]
I think screenshots is enough, if we can take screenshots at 60fps, we can create a video file from images with  ffmpeg.
https://trac.ffmpeg.org/wiki/Slideshow
